### DISCUSSION TITLE
[General] No permission to submit pr to https://github.com/matter-labs/era-zkevm_test_harness

### DISCUSSION THREAD

**Discussion by peinlcy** [zkSync-Community-Hub/zksync-developers#192] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I want to submit some pull requests to the [era-zkevm_test_harness] repository, but I don't have permission. Can someone grant me this permission?

### What is the specific issue or error you're encountering?

NA

### Can you share the error messages or logs you're receiving, if any?

NA

### Have you made any recent changes to the contract before encountering this issue?

NA

### Are there any external libraries or contracts that your contract interacts with?

NA

### Can you provide the relevant portions of your contract code where the issue is occurring?

NA

### Have you tried to isolate the problem, and if so, what were the results?

NA

### What steps have you already taken to try to resolve the issue?

NA

**Comment by MexicanAce** [comment]
Hi @peinlcy 👋

I'll look into this one with the team, but wanted to double check that you're submitting the PR from a fork branch correct? Could you also post the point at which you are receiving the error?

**Comment by peinlcy** [comment]
Yes! I have cloned the repo to my local machine, then checked out a new
branch and committed some code, when i push the new branch to
origin(matter-labs/era-zkevm_test_harness), i got this error:

```
$ git push --set-upstream origin peinlcy/sort_storage_access_queries
ERROR: Permission to matter-labs/era-zkevm_test_harness.git denied to
peinlcy.
```

**Reply by MexicanAce** [reply, marked as answer]
Access to create branches in the repo is locked down, so you will need to fork the repo and submit a Pull Request from a branch of the fork.

See here for instructions on `How to Fork a Repo` from GitHub: https://docs.github.com/en/get-started/quickstart/fork-a-repo

Thanks for reaching out and looking forward to the Pull Request 😄

**Comment by peinlcy** [comment]
Thank you, I will do that.

Nicolas Villanueva ***@***.***> 于2023年11月1日周三 21:28写道：

**Comment by peinlcy** [comment]
I found a bug in the code last week and sent an email to ***@***.***
according to the bug bounty requirement on the zkSync's official website.
However, I waited for a week and did not receive a reply, so I submitted a
PR to fix the bug.
https://github.com/matter-labs/era-zkevm_test_harness/pull/46

changyuan li ***@***.***> 于2023年11月1日周三 22:11写道：

**Comment by kimberleeni** [comment]
Sure, please provide me with your GitHub username so that I can assist you in getting the necessary permissions to submit pull requests to the [era-zkevm_test_harness] repository.

**Comment by neumagbutorcapri** [comment]
Sure, please provide your GitHub username so that we can grant you the necessary permissions to submit pull requests to the [era-zkevm_test_harness] repository.

**Comment by singletonbrittany** [comment]
Please reach out to the repository owner or maintainers to request permission to submit pull requests to the era-zkevm_test_harness repository.

**Comment by laducerbecky** [comment]
The user is requesting permission to submit pull requests to the era-zkevm_test_harness repository in the Testnet environment. They are not encountering any specific issue or error and have not made any recent changes to the contract. The user has not provided any additional details or code.

**Comment by hilliardloyd** [comment]
If you want to submit pull requests to the era-zkevm_test_harness repository but do not have permission, you can request access from the repository owner or maintainers. You can do this by opening an issue or sending a message to them directly. They will review your request and grant you permission if they deem it appropriate.
